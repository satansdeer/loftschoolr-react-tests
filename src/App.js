import React, { Component } from "react";
import ExampleComponent from "./ExampleComponent";
import "./App.css";

class App extends Component {
  state = { count: 0 };

  increment = () => {
    this.setState({ count: this.state.count + 1 });
  };

  render() {
    return (
      <div>
        <p className="count">{this.state.count}</p>
        <button onClick={this.increment}>Increment</button>
        <ExampleComponent />
      </div>
    );
  }
}

export default App;
