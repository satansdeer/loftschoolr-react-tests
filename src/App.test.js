import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders shallow", () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(".count").text()).toEqual("0");
  wrapper.find("button").simulate("click");
  expect(wrapper.find(".count").text()).toEqual("1");
});

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
