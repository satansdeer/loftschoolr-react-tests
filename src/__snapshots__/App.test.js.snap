// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p
    className="count"
  >
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <div
    className="App"
  >
    <h1>
      Example Component
    </h1>
    <p>
      Text describing example component
    </p>
  </div>
</div>
`;
